src/Umbraco.Web.UI.Client/src/common/directives/components/umbdatetimepicker.directive.js

```diff
@@ -0,0 +1,143 @@
+</pre>
+
+<h3>Controller example</h3>
+<pre>
+	(function () {
+		"use strict";
+
+		function Controller() {
+
+            var vm = this;
+
+            vm.date = "";
+
+            vm.config = {
+                pickDate: true,
+                pickTime: true,
+                useSeconds: true,
+                format: "YYYY-MM-DD HH:mm:ss",
+                icons: {
+                    time: "icon-time",
+                    date: "icon-calendar",
+                    up: "icon-chevron-up",
+                    down: "icon-chevron-down"
+                }
+            };
+
+            vm.datePickerChange = datePickerChange;
+            vm.datePickerError = datePickerError;
+
+            function datePickerChange(event) {
+                // handle change
+            }
+
+            function datePickerError(event) {
+                // handle error
+            }
+
+        }
+
+		angular.module("umbraco").controller("My.Controller", Controller);
+
+	})();
+</pre>
+
+@param {object} options (<code>binding</code>): Config object for the date picker.
+@param {string} ngModel (<code>binding</code>): Date value.
+@param {callback} onChange (<code>callback</code>): Change callback.
+@param {callback} onError (<code>callback</code>): Error callback.
+**/
+
+(function () {
+    'use strict';
+
+    function DateTimePickerDirective(assetsService) {
+
+        function link(scope, element, attrs, ctrl) {
+
+            function onInit() {
+                // load css file for the date picker
+                assetsService.loadCss('lib/datetimepicker/bootstrap-datetimepicker.min.css').then(function () {
+
+                });
+                
+                // load the js file for the date picker
+                assetsService.loadJs('lib/datetimepicker/bootstrap-datetimepicker.js').then(function () {
+                    // init date picker
+                    initDatePicker();
+                });
+            }
+
+            function onChange(event) {
+                if (scope.onChange && event.date && event.date.isValid()) {
+                    scope.$apply(function(){
+                        // Update ngModel
+                        scope.ngModel = event.date.format(scope.options.format);
+                        // callback
+                        scope.onChange({event: event});
+                    });
+                }
+            }
+
+            function onError(event) {
+                if (scope.onError) {
+                    scope.$apply(function(){
+                        // callback
+                        scope.onError({event:event});
+                    });
+                }
+            }
+
+            function initDatePicker() {
+                // Open the datepicker and add a changeDate eventlistener
+                element
+                    .datetimepicker(angular.extend({ useCurrent: true }, scope.options))
+                    .on("dp.change", onChange)
+                    .on("dp.error", onError);
```

Comment:
Does the underlying library that is being used have any other useful events that are being fired that might be useful for anyone else using this component & worth adding?